misc/metrics/Cargo.toml

```diff
@@ -32,7 +32,9 @@ prometheus-client = "0.15.0"
 libp2p-gossipsub =  { version = "0.36.0", path = "../../protocols/gossipsub", optional = true }
 
 [dev-dependencies]
+log = "0.4.0"
 env_logger = "0.8.1"
 futures = "0.3.1"
 libp2p = { path = "../../", features = ["metrics"] }
-tide = "0.16"
+hyper = { version="0.14", features = ["full"] }
+tokio = { version = "1", features = ["full"] }
```

Comment:
For the sake of reducing CI compile times, do we need `full` for each of these here?

Reply:
The problem was related to structs a couple of levels deep with missing traits implementations. I will do some research and hacking to see if the features can be narrowed.

Reply:
Cargo.toml is fixed and I just rebased.